echion/threads.h

```diff
@@ -26,6 +26,76 @@
 #include <echion/tasks.h>
 #include <echion/timing.h>
 
+
+#if defined PL_LINUX
+#define MAX_FD_COUNT 16
+
+class FileD8r
+{
+public:
+    using Ptr = std::unique_ptr<FileD8r>;
+
+    class Error : public std::exception
+    {
+    public:
+        const char *what() const noexcept override
+        {
+            return "Cannot create file descriptor object";
+        }
+    };
+
+    FileD8r(int fd) {
+        if (fd < 0)
+            throw Error();
+
+        {
+            std::lock_guard<std::mutex> lock(fd_count_lock);
+            
+            if (fd_count >= MAX_FD_COUNT) {
+                close(fd);
+                
+                throw Error();
+            }
+    
+            fd_count++;
+        }
+    
+        fd_ = fd;
+    }
+
+    ~FileD8r()
+    {
+        if (fd_ >= 0)
+            close(fd_);
+
+        {
+            std::lock_guard<std::mutex> lock(fd_count_lock);
+         
+            if (fd_count > 0)
+                fd_count--;
+        }
+    }
+
+    operator int() const { return fd_; }
+
+private:
+    int fd_ = -1;
+
+    static int fd_count;
+    static std::mutex fd_count_lock;
```

Comment:
Could this be `std::atomic<int> fd_count` instead of using a mutex? Whenever we introduce a mutex, ddtrace has to properly re-init it after `fork()`.